File: research/quantfoundry-logging/src/quantfoundry_logging/parity_harness.py
"""Parity harness — emits the same fixed log line as the Rust and TS
harnesses for the cross-runtime golden test in
``research/tests/test_logging_parity.py``."""

from __future__ import annotations

from quantfoundry_logging import get_logger, with_correlation_id


def main() -> None:
    logger = get_logger("parity-test")
    with with_correlation_id("01PARITYHARNESS0000000000A"):
        logger.info("parity.smoke", answer=42, label="fixed")


if __name__ == "__main__":
    main()

File: core/qf-logging/src/bin/parity_harness.rs
//! Parity harness — emits one fixed log line for the cross-runtime
//! golden test in `research/tests/test_logging_parity.py`.
//!
//! Outputs a single info-level event with a known `correlation_id`, event
//! name, and payload. The cross-runtime test runs this and the equivalent
//! TS / Python harnesses and asserts the parsed JSON is identical modulo
//! the `ts` field.

fn main() {
    qf_logging::init("parity-test").expect("subscriber should install");
    qf_logging::with_correlation_id("01PARITYHARNESS0000000000A", || {
        tracing::info!(event = "parity.smoke", answer = 42, label = "fixed");
    });
}
